Support for OpenCV4.1.1

kumaakh opened the pull request:
resolves https://github.com/cpp-pm/hunter/issues/4
This pull request  creates a hunter package for openCV 4.1.1

ruslo committed:
Force empty CMAKE_SYSTEM_PROCESSOR on iOS build

Support for polly iOS toolchain

ruslo committed:
Set CMAKE_CXX_FLAGS_RELEASE for iOS

Support for polly iOS toolchain

ruslo committed:
Add HunterGate module

ruslo committed:
Use HunterGate (without packages)

ruslo committed:
hunter_add_package: ZLIB, TIFF, PNG, ippicv

ruslo committed:
Add find_dependency to config (ZLIB, TIFF, PNG)

Fix: https://github.com/ruslo/hunter/issues/81

headupinclouds committed:
CMake touch ups for BUILD_opencv_world=ON

headupinclouds committed:
Fix OpenCV_LIBS for when building opencv_world

ruslo committed:
Add OPENCV_WITH_EXTRA_MODULES option

Fix issue https://github.com/ruslo/hunter/issues/167

ruslo committed:
Use Eigen from Hunter

Fix issue https://github.com/ruslo/hunter/issues/192

ruslo committed:
Add Jpeg from Hunter

ruslo committed:
Update Hunter

pretyman committed:
Added the Qt backend with static link support

When static linking, Qt resources must the initialized manually

The file static_opencv_qt_resources.cpp initializes the resources
for the highgui icons. This patch uses the QtCMakeExtra Hunter
project to add it as an interface source file.

ruslo committed:
New Eigen interface

ruslo committed:
Update Hunter

xsacha committed:
Rename ANDROID_ vars to CMAKE_ vars.

ruslo committed:
Update Hunter

ruslo committed:
Fix iOS build

pretyman committed:
Added target Qt5::Widgets to opencv_highgui target

ruslo committed:
Use Hunter to download Jpeg

ruslo committed:
Use Hunter to download Jasper

ruslo committed:
Use Hunter to download WebP

xsacha committed:
Fix CMake find_package(OpenCV) for Windows when BUILD_SHARED_LIBS is not defined.

ruslo committed:
Fix Accelerate.framework full path

headupinclouds committed:
manage ffmpeg dependency w/ hunter

This adds support for ffmpeg as a hunter package w/ appropriate options: `-DWITH_FFMPEG=ON -DOPENCV_FFMPEG_USE_FIND_PACKAGE=ON`

* update Hunter release
* update HunterGate
* enable ffmpeg package w/ `if(HUNTER_ENABLED)`
* translate ffmpeg CONFIG libs to CMake's variable style usage,
  i.e. `${FFMPEG_INCLUDE_DIRS}, ${FFMPEG_LIBRARIES}`

ruslo committed:
Update HunterGate

ruslo committed:
Update Hunter

ruslo committed:
Fix FFmpeg support

Commit:
update ffmpeg hunterizations (see detect_ffmpeg.cmake)

fix ffmpeg build test

find eigen3 dependecy automatically

fixed broken if

got rid of CHECK_MODULE and ALIASOF

Commit:
got rid of CHECK_MODULE and ALIASOF

Commit:
fixed broken if

Commit:
find eigen3 dependecy automatically

headupinclouds committed:
fix ffmpeg build test

headupinclouds committed:
update ffmpeg hunterizations (see detect_ffmpeg.cmake)

bkotzz commented on CMakeLists.txt:
```diff
@@ -103,12 +103,17 @@ if(DEFINED CMAKE_BUILD_TYPE)
-enable_testing()
+include("cmake/HunterGate.cmake")
+HunterGate(
+    URL "https://github.com/ruslo/hunter/archive/v0.23.158.tar.gz"
```
We can also use the `cpp-pm` release here https://github.com/cpp-pm/hunter/releases/tag/v0.23.214, so that people have access to the latest `OpenCV-Extra` release that you are adding in https://github.com/cpp-pm/hunter/pull/8

kumaakh replied:
Agree, I'll re run ci with this change to test if that breaks anything here.

bkotzz commented on 3rdparty/ippicv/ippicv.cmake:
```diff
@@ -38,6 +38,11 @@ function(download_ippicv root_var)
     return()
   endif()
 
+  if(HUNTER_ENABLED)
```
Do we care to wrap the other `hunter_add_package` calls in `if(HUNTER_ENABLED)`? It's not strictly necessary in the fork as the fork is meant to be used for hunter anyways

kumaakh replied:
I am a noob but I guess the main advantage of that would be if these patches were merged to upstream, they would work for both HUNTER_ENABLED and !HUNTER_ENABLED cases

rbsheth replied:
It would be nice to merge upstream, but that may or may not happen. Regardless, having the `HUNTER_ENABLED` blocks could help us patch in the future.

Commit:
point to hunter version from cpp-pm

Commit:
point to hunter version from cpp-pm

rbsheth commented on cmake/OpenCVFindLibsVideo.cmake:
```diff
@@ -19,38 +19,68 @@ if(WITH_GSTREAMER AND NOT WITH_GSTREAMER_0_10)
-      set(GSTREAMER_APP_VERSION ${ALIASOF_gstreamer-app-1.0_VERSION})
-      set(GSTREAMER_RIFF_VERSION ${ALIASOF_gstreamer-riff-1.0_VERSION})
-      set(GSTREAMER_PBUTILS_VERSION ${ALIASOF_gstreamer-pbutils-1.0_VERSION})
+#      set(GSTREAMER_BASE_VERSION ${ALIASOF_gstreamer-base-1.0_VERSION})
```
Does this break gstreamer support?

rbsheth commented on cmake/OpenCVFindLibsVideo.cmake:
```diff
@@ -19,38 +19,68 @@ if(WITH_GSTREAMER AND NOT WITH_GSTREAMER_0_10)
-    CHECK_MODULE(gstreamer-video-1.0 HAVE_GSTREAMER_VIDEO VIDEOIO)
-    CHECK_MODULE(gstreamer-app-1.0 HAVE_GSTREAMER_APP VIDEOIO)
-    CHECK_MODULE(gstreamer-riff-1.0 HAVE_GSTREAMER_RIFF VIDEOIO)
-    CHECK_MODULE(gstreamer-pbutils-1.0 HAVE_GSTREAMER_PBUTILS VIDEOIO)
```
Were these `CHECK_MODULE` macros added by a previous Hunterization?

rbsheth commented on cmake/OpenCVFindLibsVideo.cmake:
```diff
@@ -102,8 +132,9 @@ endif(WITH_GIGEAPI)
 ocv_clear_vars(HAVE_ARAVIS_API)
 if(WITH_ARAVIS)
-  check_module(glib-2.0 HAVE_ARAVIS_GLIB VIDEOIO)
+   ocv_check_modules(ARAVIS_GLIB glib-2.0)
```
Nit: remove extra space

rbsheth commented on cmake/OpenCVFindLibsVideo.cmake:
```diff
@@ -1,320 +1,3 @@
-# ----------------------------------------------------------------------------
-#  Detect 3rd-party video IO libraries
-# ----------------------------------------------------------------------------
```
We don't need this file anymore? Does this limit the `videoio` module's functionality?

kumaakh replied:
Here is a pointer to the upstream commit where this script was more or less removed https://github.com/opencv/opencv/commit/73959fed4581556d8e6b770987c819adfacceb87#diff-147a2d6c0a28c306ecf5078d84c5eb84 

I fixed my intermediate commits to now be in sync with the upstream file:
https://github.com/opencv/opencv/blob/4.1.1/cmake/OpenCVFindLibsVideo.cmake

rbsheth commented on CMakeLists.txt:
```diff
@@ -105,8 +105,8 @@ endif()
-    URL "https://github.com/ruslo/hunter/archive/v0.23.158.tar.gz"
-    SHA1 "4f3d8287a546091d1968ee80e02f121b312fceea"
+    URL "https://github.com/cpp-pm/hunter/archive/v0.23.214.tar.gz"
+    SHA1 "f55fa3a96eac57eac28e99b4c326f1fa61c685a7"
```
:+1:

Commit:
Merge branch 'hunter-4.1.1' of https://github.com/kumaakh/hunter-opencv into hunter-4.1.1

kumaakh commented:
AFAIK I have addressed all comments by updating the change set. I have also re run CI jobs to make sure nothing has been lost in transit. It will be nice to have another review.

bkotzz commented on cmake/OpenCVFindLibsGrfmt.cmake:
```diff
@@ -182,6 +229,7 @@ if(WITH_PNG)
   if(BUILD_PNG)
     ocv_clear_vars(PNG_FOUND)
   else()
+    hunter_add_package(PNG)
```
NIT: This isn't wrapped by HUNTER_ENABLED

bkotzz commented on cmake/OpenCVFindLibsGrfmt.cmake:
```diff
@@ -74,6 +87,7 @@ if(WITH_TIFF)
   if(BUILD_TIFF)
     ocv_clear_vars(TIFF_FOUND)
   else()
+    hunter_add_package(TIFF)
```
NIT: This isn't wrapped by HUNTER_ENABLED

bkotzz commented on cmake/OpenCVFindLibsGrfmt.cmake:
```diff
@@ -30,7 +31,19 @@ if(WITH_JPEG)
   if(BUILD_JPEG)
     ocv_clear_vars(JPEG_FOUND)
   else()
+    hunter_add_package(Jpeg)
```
NIT: This should be inside the `IF(HUNTER_ENABLED)` below

bkotzz commented on cmake/OpenCVFindLibsGrfmt.cmake:
```diff
@@ -6,6 +6,7 @@
 if(BUILD_ZLIB)
   ocv_clear_vars(ZLIB_FOUND)
 else()
+  hunter_add_package(ZLIB)
```
NIT: This isn't wrapped by HUNTER_ENABLED

bkotzz commented on CMakeLists.txt:
```diff
@@ -103,12 +103,17 @@ if(DEFINED CMAKE_BUILD_TYPE)
 endif()
 
-enable_testing()
+include("cmake/HunterGate.cmake")
```
NIT: This isn't wrapped by HUNTER_ENABLED

bkotzz reviewed (approved):
Looks good. If you are interested in wrapping everything with HUNTER_ENABLED such that you can try and get the patch in upstream, I've pointed out the places that are currently missed, but it's not that important that they get addressed in this fork

rbsheth reviewed (approved):
LGTM, see @bkotzz's comments if you want to try to get this upstream.

rbsheth commented:
Released: https://github.com/cpp-pm/opencv/releases/tag/v4.1.1-p0
